freeform_json_to_yaml.py

```diff
@@ -0,0 +1,440 @@
+                LOGGER.log(LogLevel.FATAL, "Error while setting up output location -- output must be a .yaml file")
+            self.output_dest = open(output_path, 'w', encoding='utf-8')
+        except Exception as e:
+            LOGGER.log(LogLevel.FATAL, "Error while setting up output location -- " + str(e))
+
+
+    def __del__(self):
+        if self.output_dest is not None:
+            self.output_dest.close()
+
+
+    def convert(self):
+        yaml_data = {}
+        yaml_data['id'] = self.json_data['scenarioData']['name']
+        yaml_data['name'] = self.json_data['scenarioData']['description']
+        sim_env = ENV_MAP[self.json_data['scene'].split('-')[1].replace('sub', 'submarine')]
+        yaml_state = {}
+        unstructured = 'TODO'
+        narrative_sections = self.json_data.get('narrative', {}).get('narrativeSections', [])
+        for x in narrative_sections:
+            if 'additionalInfo' in x:
+                # get the first additional info available for unstructured text
+                unstructured = x['additionalInfo']
+                break
+        if unstructured == 'TODO':
+            unstructured = f"You are a medic in {'a' if sim_env['type'] != 'urban' else 'an'} {sim_env['type']} environment. Several members of your team {'and some civilians ' if sim_env['type'] != 'submarine' else ''}have been injured. Please treat these casualties."
+
+        yaml_state['unstructured'] = unstructured
+        
+        
+        yaml_state['environment'] = {'sim_environment': sim_env}
+        yaml_state['supplies'] = self.get_supplies()
+        yaml_state['characters'] = self.get_characters()
+        yaml_data['state'] = yaml_state
+        yaml_scenes = self.get_freeform_scenes()
+        yaml_data['scenes'] = yaml_scenes
+        yaml.dump(yaml_data, self.output_dest, allow_unicode=True, Dumper=Indenter, sort_keys=False, default_flow_style=False)
+        LOGGER.log(LogLevel.CRITICAL_INFO, "Completed file conversion")
+
+
+    def get_supplies(self):
+        '''
+        Returns a yaml-formatted object of supplies based on the json data
+        '''
+        inventory = self.json_data['availableInventory']
+        supplies = []
+        for item in inventory:
+            supply = {"type": SUPPLY_MAP[item]}
+            if type(inventory[item]) is int:
+                supply['quantity'] = inventory[item]
+            else:
+                if inventory[item]:
+                    supply["quantity"] = 1
+                    supply['reusable'] = True
+                else:
+                    supply['quantity'] = 0 
+            supplies.append(supply)
+        return supplies
+    
+
+    def get_characters(self):
+        '''
+        Returns a yaml-formatted object of characters based on the json data
+        '''
+        characters = self.json_data['patientDataList']
+        yaml_characters = []
+        for c in characters:
+            new_c = {}
+            new_c['id'] = c['name']
```

Comment:
It's possible for two or more characters to have the same name in the JSON config (see `PI Meeting Urban.json`), which creates duplicate character ids, which of course is a validation error.  I guess you'll either need to uniquify the names (William -> William A., William B., etc.) or have a bevy of names in the tool that can be assigned to duplicate names from the JSON file.

Reply:
This seems a little overboard. Are we sure it's necessary?

Reply:
If we TA3 are never going to create characters for our open world scenarios with the same name, it might not be necessary, I would actually prefer that we actually maybe error or give a warning on the conversion if there are duplicate character names because we don't want that for TA3 experiments.

Reply:
Okay, I will error on duplicate character names. I like that solution

Reply:
Me too, FWIW.